py/objtype.c

```diff
@@ -953,6 +953,20 @@ STATIC bool check_for_special_accessors(mp_obj_t key, mp_obj_t value) {
 }
 #endif
 
+#if ENABLE_SPECIAL_ACCESSORS
+STATIC bool check_if_special_accessor(mp_obj_t key, mp_obj_t value) {
```

Comment:
`value` is unused; actually this looks like a copy of `check_for_special_accessors` with the only difference being the descriptor lookup, so it should be possible to add an argument to that function which allows it to behave like your copy. That would reduce code size.